x-pack/plugins/file_upload/public/components/json_upload_and_parse.tsx

```diff
@@ -185,13 +185,31 @@ export class JsonUploadAndParse extends Component<FileUploadComponentProps, Stat
       }),
     });
     let indexPattern;
+    let results: FileUploadGeoResults | undefined;
     try {
       indexPattern = await getIndexPatternService().createAndSave(
         {
           title: this.state.indexName,
         },
         true
       );
+      if (!indexPattern.id) {
+        throw new Error('Index pattern id not provided');
+      }
+      const geoField = indexPattern.fields.find((field) =>
+        [ES_FIELD_TYPES.GEO_POINT as string, ES_FIELD_TYPES.GEO_SHAPE as string].includes(
+          field.type
+        )
+      );
+      if (!geoField) {
+        throw new Error('geo field not created in index pattern');
+      }
+      results = {
+        indexPatternId: indexPattern.id,
+        geoFieldName: geoField.name,
+        geoFieldType: geoField.type as ES_FIELD_TYPES.GEO_POINT | ES_FIELD_TYPES.GEO_SHAPE,
+        docCount: importResults.docCount !== undefined ? importResults.docCount : 0,
```

Comment:
Could revise to: `docCount: importResults.docCount || 0` but up to you.

Reply:
I will leave as is. I don't like falsy checks with booleans and numbers since `0` is a valid number but will fail a falsy check. I think the extra language makes this statement very explicit that it takes docCount unless docCount is undefined.